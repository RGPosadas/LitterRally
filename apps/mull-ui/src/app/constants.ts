export const ROUTES = {
  HOME: '/home',
  MAP: '/map',
  NEW_EVENT: '/new-event',
  TOOLS: '/tools',
  PROFILE: '/profile',
};
