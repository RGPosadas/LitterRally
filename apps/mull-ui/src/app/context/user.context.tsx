import React from 'react';

const UserContext = React.createContext<{
  userId: number;
  setUserId: React.Dispatch<React.SetStateAction<number>>;
  accessToken: string;
  setAccessToken: React.Dispatch<React.SetStateAction<string>>;
}>(null);

export const UserProvider = UserContext.Provider;
export const UserConsumer = UserContext.Consumer;

export default UserContext;
