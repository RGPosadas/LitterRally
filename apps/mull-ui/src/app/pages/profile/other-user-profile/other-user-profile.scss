.portfolio-container {
  padding: 0 0.5rem;
  .event-card-container {
    margin-bottom: 2rem;
    &:last-child {
      margin-bottom: 1rem;
    }
  }
}

.no-events-found-msg {
  text-align: center;
}
