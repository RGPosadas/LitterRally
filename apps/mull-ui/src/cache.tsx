import { InMemoryCache, ReactiveVar, makeVar } from '@apollo/client';

export const cache: InMemoryCache = new InMemoryCache({
  typePolicies: {
    Query: {
      fields: {
        login: {
          read() {
            return isLoggedIn();
          },
        },
      },
    },
  },
});

const loginInitialValue: boolean = false;

export const isLoggedIn: ReactiveVar<boolean> = makeVar<boolean>(loginInitialValue);
