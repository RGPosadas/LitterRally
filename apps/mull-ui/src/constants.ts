import { EventRestriction, IEvent } from '@mull/types';

/**
 * enum of the current routes for the application
 */
export enum ROUTES {
  LOGIN = '/login',
  HOME = '/home',
  MAP = '/map',
  CREATE_EVENT = '/create-event',
  TOOLS = '/tools',
  MESSAGES = '/messages',
  PROFILE = '/profile',
}

export const DAY_IN_MILLISECONDS = 86400000;

export const dummyEvent: IEvent = {
  id: 1,
  title: 'Test title',
  description:
    'Lorem ipsum dolor sit amet, regione invidunt democritum vim in, movet antiopam gubergren ne per. Est nibh magna scribentur ea, vel te munere deseruisse disputationi, eros meis ludus ne sea. Odio prompta legendos in sea, ut mei scripta labores theophrastus, id molestie probatus periculis mea. Noluisse invenire splendide sed ne, at erat quando laudem nec, possim apeirian vix at. At eam animal efficiendi interpretaris, eirmod offendit adversarium per et, summo qualisque efficiendi in has. Vel detraxit accusata ea.',
  startDate: new Date(2000, 0, 1, 0, 0, 0, 0),
  endDate: new Date(2000, 0, 2, 0, 0, 0, 0),
  location: {
    id: 1,
    point: '1260 Remembrance Road Montreal, Qc',
  },
  restriction: EventRestriction.NONE,
  participants: [
    {
      name: 'User 1',
    },
    {
      name: 'User 2',
    },
  ],
  host: {
    name: 'Test user',
  },
};
