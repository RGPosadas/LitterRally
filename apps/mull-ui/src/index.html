<!DOCTYPE html>
<html lang="en">
  <head>
    <meta charset="utf-8" />
    <title>Müll</title>
    <base href="/" />

    <link rel="icon" type="image/x-icon" href="favicon.png" />
    <link rel="manifest" href="manifest.json" />

    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <meta name="description" content="Müll: Help cleanup the world!" />
    <meta name="robots" content="index, follow" />
    <meta name="theme-color" content="#27b09a" />
    <meta name="apple-mobile-web-app-capable" content="yes" />
    <meta name="apple-mobile-web-app-status-bar-style" content="default" />
    <meta name="apple-mobile-web-app-title" content="my-app" />
    <link rel="apple-touch-icon" href="assets/icons/apple-touch-icon.png" />
  </head>
  <body>
    <div id="root"></div>
    <noscript>
      <div style="font-size: 20px; display: block">
        <div>Please enable JavaScript to continue using this application.</div>
        <div>
          <a href="https://www.enable-javascript.com">https://www.enable-javascript.com</a>
        </div>
      </div>
    </noscript>
  </body>
</html>
