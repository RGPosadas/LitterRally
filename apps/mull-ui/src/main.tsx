import { ApolloClient, ApolloProvider, InMemoryCache } from '@apollo/client';
import { createUploadLink } from 'apollo-upload-client';
import React from 'react';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import App from './app/app';
import { environment } from './environments/environment';

if ('serviceWorker' in navigator) {
  window.addEventListener('load', function () {
    navigator.serviceWorker.register('/service-worker.js').then(
      function (registration) {
        // Registration was successful
        console.log('ServiceWorker registration successful with scope: ', registration.scope);
      },
      function (err) {
        // registration failed :(
        console.log('ServiceWorker registration failed: ', err);
      }
    );
  });
}

const client = new ApolloClient({
  link: createUploadLink({
    uri: `${environment.backendUrl}/graphql`,
  }),
  cache: new InMemoryCache(),
});

ReactDOM.render(
  <React.StrictMode>
    <ApolloProvider client={client}>
      <BrowserRouter>
        <App />
      </BrowserRouter>
    </ApolloProvider>
  </React.StrictMode>,
  document.getElementById('root')
);
