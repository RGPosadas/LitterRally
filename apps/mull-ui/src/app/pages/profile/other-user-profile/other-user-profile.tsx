import { ISerializedEvent } from '@mull/types';
import { History } from 'history';
import React, { useContext } from 'react';
import { useOtherUserProfileQuery, User } from '../../../../generated/graphql';
import { sortEventsByDate } from '../../../../utilities';
import { EventCard, MullBackButton } from '../../../components';
import ProfileHeader from '../../../components/profile-header/profile-header';
import UserContext from '../../../context/user.context';
import './other-user-profile.scss';

export interface OtherUserProfilePageProps {
  history: History;
  prevPage: string;
}

export const OtherUserProfilePage = ({ history, prevPage }: OtherUserProfilePageProps) => {
  const { data: otherUserProfileData, loading: userProfileLoading } = useOtherUserProfileQuery({
    variables: {
      // TODO in TASK-83: get id of other user
      id: 4,
    },
  });
  const currentUserId = useContext(UserContext).userId;

  if (userProfileLoading) return <div className="page-container">Loading...</div>;
  if (currentUserId === otherUserProfileData.user.id) history.push('profile');

  if (otherUserProfileData.portfolioEvents) {
    const events = (otherUserProfileData.portfolioEvents as unknown) as Partial<ISerializedEvent>[];
    const sortedEvents = sortEventsByDate(events);

    var eventCards = sortedEvents.map((event, index) => (
      <EventCard
        key={`user-portfolio-event-${index}`}
        isJoined={true}
        event={event}
        onClick={() => history.push(`/events/${event.id}`)}
      />
    ));
  }

  return (
    <div className="page-container">
      <MullBackButton>{prevPage}</MullBackButton>
      <ProfileHeader
        portfolioCount={otherUserProfileData.portfolioCount}
        friendCount={otherUserProfileData.friendCount}
        hostingCount={otherUserProfileData.hostingCount}
        user={otherUserProfileData.user as User}
      />

      <div className="portfolio-container" data-testid="portfolio-events">
        {eventCards}
      </div>
    </div>
  );
};

export default OtherUserProfilePage;
