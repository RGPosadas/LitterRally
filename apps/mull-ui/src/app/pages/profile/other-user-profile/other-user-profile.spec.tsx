import { MockedProvider, MockedResponse } from '@apollo/client/testing';
import '@testing-library/jest-dom/extend-expect';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import React from 'react';
import { Router } from 'react-router-dom';
import renderer, { act as actRenderer } from 'react-test-renderer';
import { ROUTES } from '../../../../../src/constants';
import {
  OtherUserProfileDocument,
  OtherUserProfileQuery,
  RelationshipType,
  UserRelationshipDocument,
  UserRelationshipQuery,
} from '../../../../generated/graphql';
import { UserProvider } from '../../../context/user.context';
import { OtherUserProfilePage } from './other-user-profile';

describe('OtherUserProfile', () => {
  const renderHelper = (history, mocks: MockedResponse[]) => {
    return (
      <UserProvider value={{ userId: 1, setUserId: jest.fn() }}>
        <MockedProvider mocks={mocks} addTypename={false}>
          <Router history={history}>
            <OtherUserProfilePage history={history} prevPage="Add Friends" />
          </Router>
        </MockedProvider>
      </UserProvider>
    );
  };

  const otherUserMock: MockedResponse = {
    request: {
      query: OtherUserProfileDocument,
      variables: {
        id: 4,
      },
    },
    result: {
      data: {
        user: {
          id: 4,
          name: 'Candice Runolfsdottir',
          description:
            "Hey, it's Candice! Thanks for following my profile~ I am really passionate about hte environment, so feel free to add me and let us go around helping the environment together!",
          joinDate: '2020-04-28T06:03:28.000Z',
          avatar: null,
        },
        friendCount: 2,
        hostingCount: 3,
        portfolioCount: 1,
        portfolioEvents: [
          {
            id: 9,
            title: 'my event',
            restriction: 0,
            startDate: '2009-06-15T13:45:30',
            endDate: '2009-07-15T13:45:30',
            location: {
              title: '51067 Schimmel Cliff, North Christop, Connecticut, 75399-7391',
            },
            image: { id: 10, mediaType: 'jpeg' },
          },
        ],
      } as OtherUserProfileQuery,
    },
  };

  it('should render successfully', () => {
    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);

    const { baseElement } = render(renderHelper(history, null));
    expect(baseElement).toBeTruthy();
  });

  it('should match snapshot', async () => {
    const mocks: MockedResponse[] = [
      otherUserMock,
      {
        request: {
          query: UserRelationshipDocument,
          variables: {
            userIdB: 4,
          },
        },
        result: {
          data: { getUserRelationship: RelationshipType.None } as UserRelationshipQuery,
        },
      },
    ];

    const history = createMemoryHistory();
    history.push(ROUTES.OTHER_USER_PROFILE);
    await actRenderer(async () => {
      const tree = renderer.create(renderHelper(history, mocks));
      await new Promise((resolve) => setTimeout(resolve, 100));
      expect(tree.toJSON()).toMatchSnapshot();
    });
  });
});
