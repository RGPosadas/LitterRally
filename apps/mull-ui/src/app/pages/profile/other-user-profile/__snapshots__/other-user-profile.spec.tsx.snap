// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OtherUserProfile should match snapshot 1`] = `
<div
  className="page-container"
>
  <button
    className="mull-back-button  "
    data-testid="mull-back-button"
    onClick={[Function]}
  >
    <svg
      aria-hidden="true"
      className="svg-inline--fa fa-chevron-left fa-w-10 mull-back-button-icon"
      data-icon="chevron-left"
      data-prefix="fas"
      focusable="false"
      role="img"
      style={Object {}}
      viewBox="0 0 320 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M34.52 239.03L228.87 44.69c9.37-9.37 24.57-9.37 33.94 0l22.67 22.67c9.36 9.36 9.37 24.52.04 33.9L131.49 256l154.02 154.75c9.34 9.38 9.32 24.54-.04 33.9l-22.67 22.67c-9.37 9.37-24.57 9.37-33.94 0L34.52 272.97c-9.37-9.37-9.37-24.57 0-33.94z"
        fill="currentColor"
        style={Object {}}
      />
    </svg>
    <div
      className="mull-back-button-text"
    >
      Add Friends
    </div>
  </button>
  <div
    className="profile-header-container"
  >
    <div
      className="user-name-container"
    >
      <h1
        className="profile-header-username"
        data-testid="userName"
      >
        Candice Runolfsdottir
      </h1>
    </div>
    <div
      className="profile-header"
    >
      <img
        alt="user"
        className="user-profile-picture"
        src="./assets/icons/icon-192x192.png"
      />
      <div
        className="profile-side-container"
      >
        <button
          className="profile-stats"
          data-testid="portfolioCount"
        >
          1
          <br />
          Portfolio
        </button>
        <button
          className="profile-stats"
          data-testid="friendCount"
        >
          1
          <br />
          Friends
        </button>
        <button
          className="profile-stats"
          data-testid="hostingCount"
        >
          3
          <br />
          Hosting
        </button>
        <div
          className="friend-status-container"
        >
          <button
            className="mull-button friend-status-button "
            data-testid="mull-button"
            onClick={[Function]}
            type="submit"
          >
            Add Friend
          </button>
        </div>
      </div>
    </div>
    <p
      className="profile-header-user-description"
      data-testid="userDescription"
    >
      Hey, it's Candice! Thanks for following my profile~ I am really passionate about hte environment, so feel free to add me and let us go around helping the environment together!
    </p>
  </div>
  <div
    className="portfolio-container"
    data-testid="portfolio-events"
  >
    <div
      className="event-card-container button"
      onClick={[Function]}
      style={Object {}}
    >
      <img
        alt="Event"
        className="event-card-image"
        src="http://localhost:3333/api/media/9"
      />
      <div
        className="event-card-datetime"
        data-testid="event-card-datetime"
      >
        <div
          className="date-style"
        >
          15 JUN
        </div>
        <div>
          01:45PM
        </div>
      </div>
      <div
        className="event-card-description"
      >
        <div
          className="event-card-text"
        >
          <div
            className="event-card-title"
          >
            my event
          </div>
          <div
            className="event-card-location"
          >
            15km â€¢ 51067 Schimmel Cliff, North Christop, Connecticut, 75399-7391
          </div>
        </div>
        <div
          className="event-members-div"
        >
          <img
            alt="participant_1"
            className="profile-picture"
            src="https://blog.photofeeler.com/wp-content/uploads/2017/04/are-bumble-profiles-fake-how-many.jpeg"
          />
          <img
            alt="participant_2"
            className="profile-picture"
            src="https://i.pinimg.com/236x/92/f0/ed/92f0edd9b0ecefdd5b7a48b8e1f7d340.jpg"
          />
          <img
            alt="participant_3"
            className="profile-picture"
            src="https://pbs.twimg.com/profile_images/748593045566853124/9DDVz0uT_400x400.jpg"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
